### PR TITLE
Cube barrier changes from GH #1033

### PR DESCRIPTION
**Pull request by Tony-LunarG**
I left in the ACCESS_MEMORY_READ bit but can easily remove it if it's decided it's wrong.  I do get validation warnings specifying a srcAccessMask with LAYOUT_UNDEFINED.

### CODE REVIEW

**Comment by szdarkhack** [comment]
The acquire barrier should not need any access masks at all, since it does not need to create a memory dependency anyway (the contents of the image need not be preserved, hence the UNDEFINED layout). I'm still unclear about the presentation barrier though.

**Comment by Tony-LunarG** [comment]
So we don't need/want to transition the image's layout to COLOR_ATTACHMENT_OPTIMAL?

**Comment by szdarkhack** [comment]
The renderpass can do the layout transitions (by setting initial layout UNDEFINED, first subpass layout COLOR_ATTACHMENT_OPTIMAL, final layout PRESENT_SRC) as per the spec in 7.1, page 126:

> The implementation will automatically perform layout transitions as needed between subpasses to make each subpass use the requested layouts.

Since we ONLY need to do layout transition after the image is acquired (i.e. no memory dependency and no ownership transfer), the renderpass can take care of that by itself and should require no extra barrier. Do note that this is not the case for presentation, as we (potentially) need to pass the image over to the presentation queue, which can only be done via a pipeline barrier.

So, to summarize, the acquire barrier should be dropped and the initial layout for both attachments in the renderpass description should be switched to UNDEFINED since we don't care about the initial contents.

**Comment by Tony-LunarG** [comment]
Ok - changes pushed.

**Comment by szdarkhack** [comment]
Looks good to me. Since this is supposed to be sample code, perhaps it would be beneficial to add a comment or two explaining why there is no barrier and why the initial layout is UNDEFINED, to serve as a better tutorial.

**Comment by Tony-LunarG** [comment]
Cherry picked to master
